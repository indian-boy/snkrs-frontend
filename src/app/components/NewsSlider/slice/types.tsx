import { FunctionComponent, SVGProps } from 'react';

export interface SliderItem {
  id: number;
  title: string;
  link: {
    label: string;
    url: string;
  };
  Image: FunctionComponent<
    SVGProps<SVGSVGElement> & { title?: string | undefined }
  >;
}

export interface NewsSliceState {
  selected: SliderItem;
  items: SliderItem[];
}
