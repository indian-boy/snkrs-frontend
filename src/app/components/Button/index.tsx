import React, { memo } from 'react';
import styled, { css, ThemeProvider } from 'styled-components/macro';
import { mainTheme } from 'styles/themes/main-theme';

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  label: string;
  secondary?: boolean;
  onClick?: () => void;
}

export const Button = memo(({ onClick, label, ...props }: Props) => {
  return (
    <ThemeProvider theme={mainTheme}>
      <ButtonWrapper onClick={onClick} type="button" {...props}>
        {label}
      </ButtonWrapper>
    </ThemeProvider>
  );
});

const ButtonWrapper = styled.button<{ secondary?: boolean }>`
  ${props =>
    css`
      background-color: ${props.theme.palette.primary.default};
      color: ${props.theme.palette.secondary.default};
    `}
  ${props =>
    props.secondary &&
    css`
      background-color: ${props.theme.palette.secondary.default};
      color: ${props.theme.palette.primary.default};
    `}
`;
