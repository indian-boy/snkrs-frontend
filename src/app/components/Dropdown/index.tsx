import React, { memo, useEffect } from 'react';
import styled, { keyframes } from 'styled-components/macro';

import Theme from 'styles/themes/main-theme';

export interface Option {
  title: string;
  selected: boolean;
  key: string | number;
}

interface Props {
  label: string;
  options: Option[];
  setOptionSelected: React.Dispatch<React.SetStateAction<Option | undefined>>;
  optionSelectedState: Option | undefined;
}

export const Dropdown = memo(
  ({ label, options, optionSelectedState, setOptionSelected }: Props) => {
    const ref = React.useRef<HTMLDetailsElement>(null);

    useEffect(() => {
      if (!optionSelectedState) {
        const [firstOption] = options;
        setOptionSelected(firstOption);
      }
    }, [optionSelectedState, options, setOptionSelected]);

    const handleClick = (optionsSelected: Option) => {
      closeSummary();
      setOptionSelected(optionsSelected);
    };

    const closeSummary = () => {
      ref.current?.removeAttribute('open');
    };

    return (
      <Theme>
        <Wrapper>
          <FieldSet>
            <Legend>{optionSelectedState?.title || label}</Legend>
            <Details ref={ref}>
              <Sumary>{optionSelectedState?.title || label}</Sumary>
              <Options>
                {options.map((item, index) => {
                  return (
                    <Label key={index}>
                      <Input
                        defaultChecked={item.key === optionSelectedState?.key}
                        onClick={() => handleClick(item)}
                        type="radio"
                        name="radio"
                      />
                      <Span>{item.title}</Span>
                    </Label>
                  );
                })}
              </Options>
            </Details>
          </FieldSet>
        </Wrapper>
      </Theme>
    );
  },
);

const Wrapper = styled.div`
  font-size: 0.875rem;

  * {
    scrollbar-width: 0;
  }

  *::-webkit-scrollbar {
    background-color: ${props => props.theme.palette.secondary.default};
    width: 0.5rem;
    border-radius: 0.5rem;
  }

  *::-webkit-scrollbar-button {
    display: none;
  }

  *::-webkit-scrollbar-thumb {
    background-color: ${props => props.theme.palette.primary.default};
    border-radius: 0.5rem;
  }

  *::-webkit-scrollbar-track {
    background-color: ${props => props.theme.palette.secondary.default};
    border-radius: 0.5rem;
  }

  *::-webkit-scrollbar-track:hover {
    background-color: ${props => props.theme.palette.secondary.default};
  }

  *::-webkit-scrollbar-thumb:hover {
    border: 0.5rem solid black;
  }
`;

const FieldSet = styled.fieldset`
  border: none;
  max-width: fit-content;
  max-height: fit-content;
`;

const Legend = styled.legend`
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

const Sumary = styled.summary`
  border: none;
  padding: 0.25rem;
  position: relative;
  border-radius: 0.25rem;
  cursor: pointer;
  list-style: none;
  padding-right: 2rem;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
  color: ${props => props.theme.palette.primary.p700};
  background-color: transparent;

  ::after {
    content: '';
    border: solid black;
    border-width: 0 1px 1px 0;
    display: inline-block;
    padding: 0.25rem;
    transform: translateY(-50%) rotate(45deg);
    transition-property: transform;
    transition-timing-function: ease-in-out;
    transition-duration: 0.2s;
    right: 0;
    top: 40%;
    position: absolute;
  }

  &::-webkit-details-marker {
    display: none;
  }
`;

const Details = styled.details`
  position: relative;

  &[open] > summary {
    ::after {
      transform: rotate(225deg);
    }
  }
`;

const Label = styled.label`
  position: relative;
  cursor: pointer;
`;

const Input = styled.input`
  visibility: hidden;
  position: absolute;
  transition: all 0.2s ease 0s;
  background-color: ${props => props.theme.palette.secondary.s200};
  color: ${props => props.theme.palette.primary.default};

  :checked + span {
    background-color: ${props => props.theme.palette.primary.default};
    color: ${props => props.theme.palette.secondary.default};
  }
`;

const Span = styled.span`
  display: block;
  padding: 1rem;
  border-radius: 0.25rem;
  margin: 0.25rem 0;
  transition: all 0.2s ease 0s;

  background-color: ${props => props.theme.palette.secondary.s200};
  color: ${props => props.theme.palette.primary.p700};

  :hover {
    background-color: ${props => props.theme.palette.primary.default};
    color: ${props => props.theme.palette.secondary.default};
  }
`;

const FadeIn = keyframes`
  0%  { opacity: 0; }
  100% { opacity: 1; }
`;

const Options = styled.div`
  background-color: ${props => props.theme.palette.secondary.s200};
  border: none;
  padding: 0.25rem;
  overflow-y: auto;
  max-height: 8rem;
  box-shadow: 0 0.2rem 0.5rem rgb(0 0 0 / 10%);
  border-radius: 0.25rem;
  animation: ${FadeIn} 0.4s ease-in-out;
`;
