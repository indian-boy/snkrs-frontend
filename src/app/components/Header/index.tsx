import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';
import { BrowserRouter } from 'react-router-dom';
import Theme from 'styles/themes/main-theme';
import { useCheckIsMediumScreen } from 'utils/helpers';
import { messages } from './messages';
import {
  BagIconCustomized,
  CenterWrapper,
  HamburguerMenuIconCustomized,
  HeaderLink,
  HeaderSearchInput,
  LeftSideWrapper,
  NikeLogoCustomized,
  RightSideWrapper,
  Wrapper,
} from './styles';

interface Props {}

export const Header = memo((props: Props) => {
  const { t } = useTranslation();
  const isMediumScreen = useCheckIsMediumScreen();

  return (
    <BrowserRouter>
      <Theme>
        <Wrapper>
          <LeftSideWrapper>
            <NikeLogoCustomized />
          </LeftSideWrapper>
          <CenterWrapper>
            <HeaderLink to="/">
              {t(messages.i18nNewOffersPageLink())}
            </HeaderLink>
            <HeaderLink to="/">{t(messages.i18nMalePageLink())}</HeaderLink>
            <HeaderLink to="/">{t(messages.i18nFemalePageLink())}</HeaderLink>
            <HeaderLink to="/">{t(messages.i18nKidsPageLink())}</HeaderLink>
            <HeaderLink to="/">{t(messages.i18nOffersPageLink())}</HeaderLink>
            <HeaderLink to="/">
              {t(messages.i18nNewOffersPageLink())}
            </HeaderLink>
            <HeaderLink to="/">{t(messages.i18nHomePageLink())}</HeaderLink>
          </CenterWrapper>
          <RightSideWrapper>
            <HeaderSearchInput
              hidden={!isMediumScreen}
              style={{ width: '12rem' }}
              secondary
              isRounded={true}
              showSearchIcon={true}
              type="search"
              placeholder={t(messages.i18nSearchButtonLabel())}
            />
            <BagIconCustomized />
            {!isMediumScreen && <HamburguerMenuIconCustomized />}
          </RightSideWrapper>
        </Wrapper>
      </Theme>
    </BrowserRouter>
  );
});
