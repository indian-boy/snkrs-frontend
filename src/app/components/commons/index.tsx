export * from './Button';
export * from './Dropdown';
export * from './LinksDropdown';
export * from './SearchInput';
