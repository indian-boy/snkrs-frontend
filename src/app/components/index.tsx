export * from './commons';
export * from './Footer/';
export * from './Header';
export * from './MapsIframeModal';
export * from './NewsLetter/';
export * from './NewsSlider';
export * from './ShoppingStoreCard';
export * from './ShoppingStoresList';
export * from './TopHeader';
