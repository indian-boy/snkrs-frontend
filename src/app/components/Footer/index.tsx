import { LinksDropdown } from 'app/components';
import { ReactComponent as Amex } from 'assets/svgs/logos/amex.svg';
import { ReactComponent as Discover } from 'assets/svgs/logos/discover.svg';
import { ReactComponent as Elo } from 'assets/svgs/logos/elo.svg';
import { ReactComponent as FacebookLogo } from 'assets/svgs/logos/facebook_rounded_white.svg';
import { ReactComponent as Hipercard } from 'assets/svgs/logos/hipercard.svg';
import { ReactComponent as InstagramLogo } from 'assets/svgs/logos/instagram_rounded_white.svg';
import { ReactComponent as Mastercard } from 'assets/svgs/logos/mastercard.svg';
import { ReactComponent as Vista } from 'assets/svgs/logos/visa.svg';
import { ReactComponent as YoutubeLogo } from 'assets/svgs/logos/youtube_rounded_white.svg';
import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';
import { BrowserRouter, NavLink } from 'react-router-dom';
import Theme from 'styles/themes/main-theme';
import { useCheckIsMediumScreen } from 'utils/helpers';
import { messages } from './messages';
import {
  CompanyInfos,
  Hr,
  Links,
  LinksMediumScreen,
  LinksSection,
  LinksWrapper,
  MainNavLink,
  MediumScreenWrapper,
  MultipleSectionsWrapper,
  PaymentMethods,
  PaymentOptions,
  SectionLabel,
  SectionNavLink,
  SectionTitleMediumScreen,
  SmallScreenWrapper,
  SocialMedias,
  SocialMediasOptions,
  TermLink,
  TermsLinksMediumScreen,
  TermsSectionMediumScreen,
  TermsSmallScreen,
} from './styles';

interface Props {}

export const Footer = memo((props: Props) => {
  const { t } = useTranslation();

  const helpLinks = [
    {
      label: t(messages.i18nHelpLinks.links.i18nGeneralQuestions()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18FindYourSize()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nOrders()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nExchangesAndReturns()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nEditConsent()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nPayments()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nCookiePreferences()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nProducts()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nCorporate()),
      url: '/',
    },
    {
      label: t(messages.i18nHelpLinks.links.i18nContactUs()),
      url: '/',
    },
  ];

  const aboutNikeLinks = [
    {
      label: t(messages.i18nAboutNikeLinks.links.i18nMadeToPlay()),
      url: '/',
    },
    {
      label: t(messages.i18nAboutNikeLinks.links.i18nGeneralSustentability()),
      url: '/',
    },
  ];

  const isMediumScreen = useCheckIsMediumScreen();

  return (
    <BrowserRouter>
      <Theme>
        {isMediumScreen && (
          <LinksSection>
            <MediumScreenWrapper>
              <LinksMediumScreen>
                <MainNavLink to="/">
                  {t(messages.i18nFindShoppingStore())}
                </MainNavLink>
                <MainNavLink to="/">
                  {t(messages.i18nRegisterForNews())}
                </MainNavLink>
                <MainNavLink to="/">{t(messages.i18nSiteMap())}</MainNavLink>
              </LinksMediumScreen>
              <LinksMediumScreen>
                <SectionTitleMediumScreen>
                  {t(messages.i18nHelpLinks.i18nTitle())}
                </SectionTitleMediumScreen>
                {helpLinks &&
                  helpLinks.map((helpLink, index) => (
                    <SectionNavLink key={index} to={helpLink.url}>
                      {helpLink.label}
                    </SectionNavLink>
                  ))}
              </LinksMediumScreen>
              <LinksMediumScreen>
                <SectionTitleMediumScreen>
                  {t(messages.i18nAboutNikeLinks.i18nTitle())}
                </SectionTitleMediumScreen>
                {aboutNikeLinks &&
                  aboutNikeLinks.map((aboutNikeLink, index) => (
                    <SectionNavLink key={index} to={aboutNikeLink.url}>
                      {aboutNikeLink.label}
                    </SectionNavLink>
                  ))}
              </LinksMediumScreen>
              <MultipleSectionsWrapper>
                <SocialMedias>
                  <SectionLabel>{t(messages.i18nSocialMedias())}</SectionLabel>
                  <SocialMediasOptions>
                    <NavLink to="/">
                      <FacebookLogo></FacebookLogo>
                    </NavLink>
                    <NavLink to="/">
                      <InstagramLogo></InstagramLogo>
                    </NavLink>
                    <NavLink to="/">
                      <YoutubeLogo></YoutubeLogo>
                    </NavLink>
                  </SocialMediasOptions>
                </SocialMedias>
                <PaymentMethods>
                  <SectionLabel>
                    {t(messages.i18nPaymentMethods())}
                  </SectionLabel>
                  <PaymentOptions>
                    <Amex></Amex>
                    <Discover></Discover>
                    <Elo></Elo>
                    <Hipercard></Hipercard>
                    <Mastercard></Mastercard>
                    <Vista></Vista>
                  </PaymentOptions>
                </PaymentMethods>
              </MultipleSectionsWrapper>
            </MediumScreenWrapper>
            <Hr />
            <TermsSectionMediumScreen>
              <TermsLinksMediumScreen>
                <TermLink to="/">Brasil</TermLink>
                <TermLink to="/">{t(messages.i18nPrivacyPolicy())}</TermLink>
                <TermLink to="/">{t(messages.i18nUserTerms())}</TermLink>
              </TermsLinksMediumScreen>
              <CompanyInfos>{t(messages.i18nCompanyInfos())}</CompanyInfos>
            </TermsSectionMediumScreen>
          </LinksSection>
        )}

        {!isMediumScreen && (
          <SmallScreenWrapper>
            <Links>
              <MainNavLink to="/">
                {t(messages.i18nFindShoppingStore())}
              </MainNavLink>
              <MainNavLink to="/">
                {t(messages.i18nRegisterForNews())}
              </MainNavLink>
              <MainNavLink to="/">{t(messages.i18nSiteMap())}</MainNavLink>
            </Links>
            <Hr />
            <LinksWrapper>
              <LinksDropdown
                title={t(messages.i18nHelpLinks.i18nTitle())}
                links={helpLinks}
              ></LinksDropdown>
              <LinksDropdown
                title={t(messages.i18nAboutNikeLinks.i18nTitle())}
                links={aboutNikeLinks}
              ></LinksDropdown>
            </LinksWrapper>
            <SocialMedias>
              <SectionLabel>{t(messages.i18nSocialMedias())}</SectionLabel>
              <SocialMediasOptions>
                <NavLink to="/">
                  <FacebookLogo></FacebookLogo>
                </NavLink>
                <NavLink to="/">
                  <InstagramLogo></InstagramLogo>
                </NavLink>
                <NavLink to="/">
                  <YoutubeLogo></YoutubeLogo>
                </NavLink>
              </SocialMediasOptions>
            </SocialMedias>
            <PaymentMethods>
              <SectionLabel>{t(messages.i18nPaymentMethods())}</SectionLabel>
              <PaymentOptions>
                <Amex></Amex>
                <Discover></Discover>
                <Elo></Elo>
                <Hipercard></Hipercard>
                <Mastercard></Mastercard>
                <Vista></Vista>
              </PaymentOptions>
            </PaymentMethods>
            <Hr />
            <TermsSmallScreen>
              <TermLink to="/">Brasil</TermLink>
              <TermLink to="/">{t(messages.i18nPrivacyPolicy())}</TermLink>
              <TermLink to="/">{t(messages.i18nUserTerms())}</TermLink>
            </TermsSmallScreen>
            <CompanyInfos>{t(messages.i18nCompanyInfos())}</CompanyInfos>
          </SmallScreenWrapper>
        )}
      </Theme>
    </BrowserRouter>
  );
});
