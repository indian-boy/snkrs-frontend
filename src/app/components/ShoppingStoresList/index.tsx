import { Dropdown, Option } from 'app/components';
import React, { Dispatch, memo, SetStateAction, useState } from 'react';
import { useTranslation } from 'react-i18next';
import Theme from 'styles/themes/main-theme';
import { ShoppingStore } from 'types';
import { ShoppingStoreCard } from '../ShoppingStoreCard';
import { messages } from './messages';
import { Filters, List, Wrapper } from './styles';

interface Props {
  setShoppingStoreSelected: Dispatch<SetStateAction<ShoppingStore | null>>;
  shoppingStores: ShoppingStore[];
}

export const ShoppingStoresList = memo(
  ({ setShoppingStoreSelected, shoppingStores, ...props }: Props) => {
    const { t } = useTranslation();

    const options = [
      { key: 1, title: t(messages.i18nMinorDistance()) },
      { key: 2, title: t(messages.i18nGreaterDistance()) },
    ];

    const [optionSelectedState, setOptionSelected] = useState<Option>(
      options[0],
    );

    const onSelectShoppingStore = value => {
      setShoppingStoreSelected({ ...value });
    };

    return (
      <Theme>
        <Wrapper>
          <Filters>
            <Dropdown
              label={t(messages.i18nFilter())}
              options={options}
              optionSelectedState={optionSelectedState}
              setOptionSelected={setOptionSelected}
            />
          </Filters>
          <List>
            {shoppingStores &&
              shoppingStores.map((store, index) => (
                <ShoppingStoreCard
                  onClick={() => onSelectShoppingStore(store)}
                  {...store}
                  key={index}
                ></ShoppingStoreCard>
              ))}
          </List>
        </Wrapper>
      </Theme>
    );
  },
);
