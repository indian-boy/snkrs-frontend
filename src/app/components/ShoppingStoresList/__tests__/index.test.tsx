import { render } from '@testing-library/react';
import { Option } from 'app/components';
import React, { useState } from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import { configureAppStore } from 'store/configureStore';
import { CommonUsedAttributes, ShoppingStore } from 'types';
import { ShoppingStoreProps, ShoppingStoresList } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const shoppingStoresMock: ShoppingStore[] = [
  {
    id: 1,
    placeName: 'Avenida paulista',
    address: {
      latitude: -23.5639655,
      longitude: -46.6560802,
      location: 'Avenida Paulista, 1227 - Bela Vista São Paulo - SP, 01311-200',
    },
    availability: 4,
    serviceHours: 'Segunda a Sábado 10h às 22h | Domingo 11h às 20h',
    distance: 1000,
  },
  {
    id: 2,
    placeName: 'Regus',
    address: {
      latitude: -23.5469643,
      longitude: -46.7624297,
      location:
        'Rua Werner Von Siemens, 111 - Lapa de Baixo, São Paulo - SP, 05069-010',
    },
    availability: 6,
    serviceHours: 'Segunda a Sábado 10h às 22h | Domingo 11h às 20h',
    distance: 1000,
  },
];

const ParentComponentRenderingWithProviders = (
  props: Partial<ShoppingStoreProps> & CommonUsedAttributes,
  store: Store,
) => {
  const { shoppingStores } = props;

  const [, setShoppingStoreSelected] = React.useState<ShoppingStore | null>(
    shoppingStoresMock[0],
  );

  const filterOptions = [
    { key: 1, title: 'Filter #1' },
    { key: 2, title: 'Filter #2' },
  ];

  const [filterOptionSelectedState, setFilterOptionSelected] = useState<
    Option | undefined
  >(filterOptions[0]);

  return (
    <Provider store={store}>
      <ShoppingStoresList
        {...props}
        shoppingStores={shoppingStores ?? []}
        filterOptions={filterOptions}
        filterOptionSelectedState={filterOptionSelectedState}
        setFilterOptionSelected={setFilterOptionSelected}
        setShoppingStoreSelected={setShoppingStoreSelected}
      />
    </Provider>
  );
};

describe('<ShoppingStoresList  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const ParentComponent = () =>
      ParentComponentRenderingWithProviders(
        {
          'data-testid': 'shoppingStoresListID',
          shoppingStores: shoppingStoresMock,
        },
        store,
      );

    const { getByTestId, container } = render(<ParentComponent />);
    const shoppingStoresListElement = getByTestId('shoppingStoresListID');

    expect(shoppingStoresListElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
