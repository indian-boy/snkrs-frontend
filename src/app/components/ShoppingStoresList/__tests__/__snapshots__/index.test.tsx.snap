// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingStoresList  /> should match snapshot 1`] = `
.c2 {
  font-size: 0.875rem;
  display: contents;
}

.c2 * {
  -webkit-scrollbar-width: 0;
  -moz-scrollbar-width: 0;
  -ms-scrollbar-width: 0;
  scrollbar-width: 0;
}

.c2 *::-webkit-scrollbar {
  background-color: #FFFFFF;
  width: 0.5rem;
  border-radius: 0.5rem;
}

.c2 *::-webkit-scrollbar-button {
  display: none;
}

.c2 *::-webkit-scrollbar-thumb {
  background-color: #000000;
  border-radius: 0.5rem;
}

.c2 *::-webkit-scrollbar-track {
  background-color: #FFFFFF;
  border-radius: 0.5rem;
}

.c2 *::-webkit-scrollbar-track:hover {
  background-color: #FFFFFF;
}

.c2 *::-webkit-scrollbar-thumb:hover {
  border: 0.5rem solid black;
}

.c3 {
  border: none;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  max-height: -webkit-fit-content;
  max-height: -moz-fit-content;
  max-height: fit-content;
}

.c4 {
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c6 {
  border: none;
  padding: 0.25rem;
  position: relative;
  border-radius: 0.25rem;
  cursor: pointer;
  list-style: none;
  padding-right: 2rem;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
  color: #1A1A1A;
  background-color: transparent;
}

.c6::after {
  content: '';
  border: solid black;
  border-width: 0 1px 1px 0;
  display: inline-block;
  padding: 0.25rem;
  -webkit-transform: translateY(-50%) rotate(45deg);
  -ms-transform: translateY(-50%) rotate(45deg);
  transform: translateY(-50%) rotate(45deg);
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  right: 0;
  top: 40%;
  position: absolute;
}

.c6::-webkit-details-marker {
  display: none;
}

.c5 {
  position: relative;
}

.c5[open] > summary::after {
  -webkit-transform: rotate(225deg);
  -ms-transform: rotate(225deg);
  transform: rotate(225deg);
}

.c8 {
  position: relative;
  cursor: pointer;
}

.c9 {
  visibility: hidden;
  position: absolute;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  background-color: #F7F7F7;
  color: #000000;
}

.c9:checked + span {
  background-color: #000000;
  color: #FFFFFF;
}

.c10 {
  display: block;
  padding: 1rem;
  border-radius: 0.25rem;
  margin: 0.25rem 0;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  background-color: #F7F7F7;
  color: #1A1A1A;
}

.c10:hover {
  background-color: #000000;
  color: #FFFFFF;
}

.c7 {
  position: absolute;
  width: 100%;
  background-color: #F7F7F7;
  border: none;
  padding: 0.25rem;
  overflow-y: auto;
  max-height: 8rem;
  box-shadow: 0 0.2rem 0.5rem rgb(0 0 0 / 10%);
  border-radius: 0.25rem;
  -webkit-animation: llfyzB 0.4s ease-in-out;
  animation: llfyzB 0.4s ease-in-out;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  font-weight: 500;
  font-size: 1.5rem;
  line-height: 1.2rem;
  color: #1A1A1A;
}

.c14 {
  font-weight: 500;
  font-size: 1.125rem;
  color: #1A1A1A;
}

.c15 {
  background-color: transparent;
  font-style: normal;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
  cursor: pointer;
  outline: none;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  color: #1A1A1A;
  margin-top: 0.5rem;
  padding: 0;
}

.c15:focus,
.c15:active {
  outline: none;
}

.c16 {
  background-color: #FFFFFF;
  margin: 1rem 0;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 1.5rem;
  gap: 0.5rem;
}

.c17 {
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #737373;
}

.c18 {
  font-style: normal;
  font-weight: bold;
  font-size: 0.75rem;
  line-height: 1.5rem;
  color: #006611;
}

.c0 {
  padding: 1rem;
  border-radius: 0.5rem;
  background-color: #F7F7F7;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <fieldset
        class="c3"
      >
        <legend
          class="c4"
        >
          Option #1
        </legend>
        <details
          class="c5"
        >
          <summary
            class="c6"
          >
            Option #1
          </summary>
          <div
            class="c7"
          >
            <label
              class="c8"
            >
              <input
                checked=""
                class="c9"
                name="radio"
                type="radio"
              />
              <span
                class="c10"
              >
                Option #1
              </span>
            </label>
            <label
              class="c8"
            >
              <input
                class="c9"
                name="radio"
                type="radio"
              />
              <span
                class="c10"
              >
                Option #2
              </span>
            </label>
          </div>
        </details>
      </fieldset>
    </div>
  </div>
  <div
    class="c11"
  >
    <div
      class="ShoppingStoreCard__Wrapper-sc-zdqw9n-0"
    >
      <div
        class="c12"
      >
        <span
          class="c13"
        >
          Avenida paulista
        </span>
        <span
          class="c14"
        >
          24km
        </span>
      </div>
      <button
        class="c15"
        type="button"
      >
        <svg>
          pin_location.svg
        </svg>
        components.shoppingStoreCard.seeOnMapLabel
        Ver no mapa
      </button>
      <div
        class="c16"
      >
        <span
          class="c17"
        >
          Avenida Paulista, 1227 - Bela Vista São Paulo - SP, 01311-200
        </span>
        <span
          class="c17"
        >
          components.shoppingStoreCard.service
          Atendimento
          :
          <br />
          Segunda a Sábado 10h às 22h | Domingo 11h às 20h
        </span>
        <span
          class="c18"
        >
          components.shoppingStoreCard.availability
          Disponível em {{count}} dias úteis
        </span>
      </div>
    </div>
    <div
      class="ShoppingStoreCard__Wrapper-sc-zdqw9n-0"
    >
      <div
        class="c12"
      >
        <span
          class="c13"
        >
          Avenida paulista
        </span>
        <span
          class="c14"
        >
          24km
        </span>
      </div>
      <button
        class="c15"
        type="button"
      >
        <svg>
          pin_location.svg
        </svg>
        components.shoppingStoreCard.seeOnMapLabel
        Ver no mapa
      </button>
      <div
        class="c16"
      >
        <span
          class="c17"
        >
          Avenida Paulista, 1227 - Bela Vista São Paulo - SP, 01311-200
        </span>
        <span
          class="c17"
        >
          components.shoppingStoreCard.service
          Atendimento
          :
          <br />
          Segunda a Sábado 10h às 22h | Domingo 11h às 20h
        </span>
        <span
          class="c18"
        >
          components.shoppingStoreCard.availability
          Disponível em {{count}} dias úteis
        </span>
      </div>
    </div>
  </div>
</div>
`;
